projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/routing_policy/statement/Statements.java

```diff
@@ -73,6 +79,41 @@ public Result execute(Environment environment) {
           result.setFallThrough(true);
           break;
 
+        case RemovePrivateAs:
+          {
+            BgpRoute.Builder bgpRouteBuilder = (BgpRoute.Builder) environment.getOutputRoute();
+            List<SortedSet<Integer>> newAsPath =
+                bgpRouteBuilder
+                    .getAsPath()
+                    .stream()
+                    .map(
+                        asSet ->
+                            asSet
+                                .stream()
+                                .filter(as -> !AsPath.isPrivateAs(as))
+                                .collect(ImmutableSortedSet.toImmutableSortedSet(Integer::compare)))
+                    .filter(asSet -> !asSet.isEmpty())
+                    .collect(ImmutableList.toImmutableList());
+            bgpRouteBuilder.setAsPath(newAsPath);
+            if (environment.getWriteToIntermediateBgpAttributes()) {
+              BgpRoute.Builder ir = environment.getIntermediateBgpAttributes();
+              List<SortedSet<Integer>> iAsPath =
+                  ir.getAsPath()
+                      .stream()
+                      .map(
+                          asSet ->
+                              asSet
+                                  .stream()
+                                  .filter(AsPath::isPrivateAs)
```

Comment:
is this logic supposed to be negated from L93?

assuming not, and the same behavior is intended to be applied to both lists, extract this to a private static helper function and then use it twice.